src/measure_theory/interval_integral.lean

```diff
@@ -907,18 +907,18 @@ lemma continuous_on_primitive {f : α → E} {a b : α} [has_no_atoms μ]
   (h_int : integrable_on f (Icc a b) μ) :
   continuous_on (λ x, ∫ t in Ioc a x, f t ∂ μ) (Icc a b) :=
 begin
-  cases le_or_lt a b with H H,
-  { have : ∀ x ∈ Icc a b, ∫ (t : α) in Ioc a x, f t ∂μ = ∫ (t : α) in a..x, f t ∂μ,
-    { intros x x_in,
-      simp_rw [← interval_oc_of_le H, integral_of_le x_in.1] },
-    rw continuous_on_congr this,
-    intros x₀ hx₀,
-    refine continuous_within_at_primitive (measure_singleton x₀) _,
-    rw interval_integrable_iff,
-    simp only [H, max_eq_right, min_eq_left],
-    exact h_int.mono Ioc_subset_Icc_self le_rfl },
-  { rw Icc_eq_empty H,
-    apply continuous_on_empty },
+  by_cases h : a ≤ b, swap,
+  { rw Icc_eq_empty h,
+    exact continuous_on_empty _ },
+  have : ∀ x ∈ Icc a b, ∫ (t : α) in Ioc a x, f t ∂μ = ∫ (t : α) in a..x, f t ∂μ,
+  { intros x x_in,
+    simp_rw [← interval_oc_of_le h, integral_of_le x_in.1] },
+  rw continuous_on_congr this,
+  intros x₀ hx₀,
+  refine continuous_within_at_primitive (measure_singleton x₀) _,
+  rw interval_integrable_iff,
+  simp only [h, max_eq_right, min_eq_left],
+  exact h_int.mono Ioc_subset_Icc_self le_rfl,
```

Comment:
Is the only change here switching the order of the goals to work on the shorter one first? If so I don't think I'm in favor of this change.

Reply:
I changed `le_or_lt a b` by `by_cases h : a ≤ b`. But I can remove the swap if you want.

Reply:
I think removing the `swap` to keep the diff small is a good idea.